server/parsec/asgi/__init__.py

```diff
@@ -41,6 +44,23 @@
     },
 ]
 
+client_base_url = "/client"
+
+
+class SPARedirectMiddleware(BaseHTTPMiddleware):
```

Comment:
Instead of providing a middleware that gets called for every route, we can instead just overload the `StaticFiles` class:

```python
class StaticFilesWithSPARedirect(StaticFiles):
    def lookup_path(self, path: str) -> tuple[str, os.stat_result | None]:
        match super().lookup_path(path):
            case (_, None):
                return super().lookup_path("index.html")
            case found:
                return found
```
(I've quickly tested this code, it seems to work as intended)

Reply:
Sees better than my solution for embedded app

Reply:
Test ok on my server !
done.